### PR TITLE
[Bug][Starter] Fix variables cannot be set in flink mode

### LINKED ISSUE #1851: Report error when using variable
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

when i use variable to start flink job, than report an error

### SeaTunnel Version

2.1.1

### SeaTunnel Config

```conf
transform {
  sql {
    sql = "select * from trans_slot where container_type = '${container_type}'"
    result_table_name = "trans_slot_new"
  }
}
```

### Running Command

```shell
./bin/start-seatunnel-flink.sh -c ~/data_process_test/debug_job/variable_2.conf --variable container_type=5
```

### Error Exception

```log
The program finished with the following exception:

org.apache.flink.client.program.ProgramInvocationException: The main method caused an error: Was passed main parameter '-Dcontainer_type=5' but no main parameter was defined in your arg class
        at org.apache.flink.client.program.PackagedProgram.callMainMethod(PackagedProgram.java:372)
        at org.apache.flink.client.program.PackagedProgram.invokeInteractiveModeForExecution(PackagedProgram.java:222)
        at org.apache.flink.client.ClientUtils.executeProgram(ClientUtils.java:114)
        at org.apache.flink.client.cli.CliFrontend.executeProgram(CliFrontend.java:812)
        at org.apache.flink.client.cli.CliFrontend.run(CliFrontend.java:246)
        at org.apache.flink.client.cli.CliFrontend.parseAndRun(CliFrontend.java:1054)
        at org.apache.flink.client.cli.CliFrontend.lambda$main$10(CliFrontend.java:1132)
        at java.security.AccessController.doPrivileged(Native Method)
        at javax.security.auth.Subject.doAs(Subject.java:422)
        at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1836)
        at org.apache.flink.runtime.security.contexts.HadoopSecurityContext.runSecured(HadoopSecurityContext.java:41)
        at org.apache.flink.client.cli.CliFrontend.main(CliFrontend.java:1132)
Caused by: com.beust.jcommander.ParameterException: Was passed main parameter '-Dcontainer_type=5' but no main parameter was defined in your arg class
        at com.beust.jcommander.JCommander.initMainParameterValue(JCommander.java:954)
        at com.beust.jcommander.JCommander.parseValues(JCommander.java:755)
        at com.beust.jcommander.JCommander.parse(JCommander.java:356)
        at com.beust.jcommander.JCommander.parse(JCommander.java:335)
        at org.apache.seatunnel.utils.CommandLineUtils.parseFlinkArgs(CommandLineUtils.java:44)
        at org.apache.seatunnel.SeatunnelFlink.main(SeatunnelFlink.java:26)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.flink.client.program.PackagedProgram.callMainMethod(PackagedProgram.java:355)
        ... 11 more
```

### Flink or Spark Version

1.13.6

### Java or Scala Version

1.8

### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of this pull request
close #1851

We cannot use `-D` to set the JVM args in flink run script, this PR will use `export` to set the JVM args.

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by ruanwenjun** [comment]
@legendtkl  @BenJFan Take a look if you are interested.

File: seatunnel-core/seatunnel-core-flink/src/main/java/org/apache/seatunnel/core/flink/FlinkEnvParameterParser.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+public class FlinkEnvParameterParser {
+
+    @SuppressWarnings("checkstyle:RegexpSingleline")
+    public static void main(String[] args) throws FileNotFoundException {
+        FlinkCommandArgs flinkCommandArgs = CommandLineUtils.parseCommandArgs(args, FlinkJobType.JAR);
+        List<String> envParameters = getEnvParameters(flinkCommandArgs);
+        System.out.println(String.join(" ", envParameters));
+    }
+
+    static List<String> getEnvParameters(FlinkCommandArgs flinkCommandArgs) throws FileNotFoundException {
```
Comment (legendtkl) [inline seatunnel-core/seatunnel-core-flink/src/main/java/org/apache/seatunnel/core/flink/FlinkEnvParameterParser.java:45]: I think a UT for `getEnvParameters` is necessary, just like `org.apache.seatunnel.core.sql.FlinkEnvParameterParserTest`

File: seatunnel-core/seatunnel-core-flink/src/main/java/org/apache/seatunnel/core/flink/utils/CommandLineUtils.java [hunk 2]
```diff
@@ -71,19 +67,6 @@ public static List<String> buildFlinkCommand(FlinkCommandArgs flinkCommandArgs,
         if (flinkCommandArgs.isCheckConfig()) {
             command.add("--check");
         }
-        // set System properties
-        flinkCommandArgs.getVariables().stream()
-          .filter(Objects::nonNull)
-          .map(String::trim)
-          .forEach(variable -> command.add("-D" + variable));
-
-        if (jobType.equals(FlinkJobType.JAR)) {
```
Comment (legendtkl) [inline seatunnel-core/seatunnel-core-flink/src/main/java/org/apache/seatunnel/core/flink/utils/CommandLineUtils.java:80]: The function parameter `jobType` could be removed.

**Review by legendtkl — COMMENTED** [review]
LGTM with minor comments.

File: seatunnel-core/seatunnel-core-flink/src/main/java/org/apache/seatunnel/core/flink/FlinkEnvParameterParser.java (hunk 1 above)
Comment (ruanwenjun) [reply seatunnel-core/seatunnel-core-flink/src/main/java/org/apache/seatunnel/core/flink/FlinkEnvParameterParser.java:45]: Done.

File: seatunnel-core/seatunnel-core-flink/src/main/java/org/apache/seatunnel/core/flink/utils/CommandLineUtils.java (hunk 2 above)
Comment (ruanwenjun) [reply seatunnel-core/seatunnel-core-flink/src/main/java/org/apache/seatunnel/core/flink/utils/CommandLineUtils.java:80]: Done.

**Review by Hisoka-X — APPROVED** [review]
LGTM

**Comment by legendtkl** [comment]
LGTM

**Review by CalvinKirs — APPROVED** [review]

**Review by wuchunfu — APPROVED** [review]
LGTM

**Pull request merged**